### ISSUE TITLE
Add support of `GROUPING SETS` to `Athena` dialect

### ISSUE THREAD

**Issue by KulykDmytro** [sqlfluff/sqlfluff#4860]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Athena dialect causing `PRS` violation when met with `GROUPING SETS`
query:
```sql
select
    as_of_date,
    channel,
    sum(total_bet_count) as cnt
from aggregate.bet_aggregate_total
where as_of_date >= date '2023-05-05'
group by grouping sets ((as_of_date, channel), (as_of_date), ())
order by as_of_date, channel
```
cli output:
```verilog
dmytro.kulyk@MB-DAT-564087 datalake % sqlfluff lint --ignore-local-config --dialect athena --verbose sandbox/grouping_sets.sql
==== sqlfluff ====
sqlfluff:                2.1.0 python:                3.10.11
implementation:        cpython verbosity:                   1
dialect:                athena templater:               jinja
rules:                                all
==== readout ====

== [sandbox/grouping_sets.sql] FAIL                                                                                                                                                            
L:   6 | P:  38 |  PRS | Line 6, Position 38: Found unparsable section: '\ngroup
                       | by grouping sets ((as_of_date, ch...'
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
==== summary ====
violations:        1 status:         FAIL
```

### Use case

Implementation can be get from `postgres` dialect

### Dialect

athena

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by KulykDmytro** [comment]
As soon as `CUBE` / `ROLLUP` / `GROUPING SETS` has been added into `SQL-2016` is it make sense to add this into `ANSI` dialect globally?
